File: dashboard/frontend/src/types/inventory.ts
export interface ResourceItem {
  resource_id: string
  resource_name: string
  service_type: string
  environment: string
  owner: string
  status: string
  region: string
  monthly_cost: number
  usage_profile: string
  created_date: string
  tags: Record<string, string>
}

export interface InventorySummary {
  total_resources: number
  by_service_type: Record<string, number>
  by_status: Record<string, number>
  by_environment: Record<string, number>
  by_owner: Record<string, number>
  total_monthly_cost: number
  cost_by_service_type: Record<string, number>
}

File: dashboard/backend/app/schemas/inventory.py
from pydantic import BaseModel


class ResourceItem(BaseModel):
    resource_id: str
    resource_name: str
    service_type: str
    environment: str
    owner: str
    status: str
    region: str
    monthly_cost: float
    usage_profile: str
    created_date: str
    tags: dict = {}


class InventorySummary(BaseModel):
    total_resources: int
    by_service_type: dict[str, int]
    by_status: dict[str, int]
    by_environment: dict[str, int]
    by_owner: dict[str, int]
    total_monthly_cost: float
    cost_by_service_type: dict[str, float]
